### PR TITLE
fixes #3 validate invalid nesting of IfStatements

### LINKED ISSUE #3: validate invalid nesting of IfStatements
Tests from esvalid: https://github.com/estools/esvalid/blob/a57c8c48a335d10eb7e1554c15c887c04103174e/test/unit.js#L71-L81

Implementation from esvalid: https://github.com/estools/esvalid/blob/a57c8c48a335d10eb7e1554c15c887c04103174e/index.js#L402

`isProblematicIfStatement` implementation: https://github.com/estools/esutils/blob/a858d2d8badd6f5a4fdfc2d46c3499608a3bf06d/lib/ast.js#L112

### PR DESCRIPTION
**Pull request by emattson**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -44,6 +44,43 @@ function isIterationStatement(type) {
...
+  }
+  let current = node.consequent;
+  do {
+    if (current.type === "IfStatement") {
+      if (current.alternate == null) {
+        return true;
+      }
+    }
+    current = trailingStatement(current);
+  } while(current);
```
Comment (michaelficarra) [inline src/index.js]: `current != null`

File: src/index.js [hunk 2]
```diff
@@ -44,6 +44,43 @@ function isIterationStatement(type) {
...
+    return node.consequent;
+
+  case "LabeledStatement":
+  case "ForStatement":
+  case "ForInStatement":
+  case "WhileStatement":
+  case "WithStatement":
+    return node.body;
+  }
+    return null;
```
Comment (michaelficarra) [inline src/index.js]: too many indents

File: src/index.js [hunk 3]
```diff
@@ -44,6 +44,43 @@ function isIterationStatement(type) {
...
+  if (node.type !== "IfStatement") {
+    return false;
+  }
+  if (node.alternate == null) {
+    return false;
+  }
+  let current = node.consequent;
+  do {
+    if (current.type === "IfStatement") {
+      if (current.alternate == null) {
```
Comment (michaelficarra) [inline src/index.js]: Combine with above test using `&&`.

File: test/unit.js [hunk 4]
```diff
@@ -154,6 +154,19 @@ suite("unit", () => {
     invalidStmt(1, new Shift.FunctionDeclaration(ID, [new Shift.Identifier("if"), ID], new Shift.FunctionBody([], [])));
   });
 
+  test("IfStatement with null `alternate` must not be the `consequent` of an IfStatement with a non-null `alternate`", () => {
+    validStmt(new Shift.IfStatement(EXPR, new Shift.DoWhileStatement(new Shift.IfStatement(EXPR, STMT, null), EXPR), STMT));
+    validStmt(new Shift.IfStatement(EXPR, new Shift.IfStatement(EXPR, STMT, STMT), STMT));
```
Comment (michaelficarra) [inline test/unit.js]: You're missing a valid test with a `null` alternate on the outer `IfStatement` and a consequent that would otherwise be problematic.

Comment (emattson) [inline test/unit.js]: done
